sierra_adapter/sierra_to_dynamo/src/main/scala/uk/ac/wellcome/platform/sierra_to_dynamo/services/WindowExtractor.scala

```diff
@@ -0,0 +1,39 @@
+package uk.ac.wellcome.platform.sierra_to_dynamo.services
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import com.twitter.inject.Logging
+import io.circe.Json
+import io.circe.optics.JsonPath.root
+import io.circe.parser.parse
+import uk.ac.wellcome.sqs.SQSReaderGracefulException
+
+import scala.util.Try
+
+object WindowExtractor extends Logging{
+  def extractWindow(jsonString: String): Try[String] = Try(parse(jsonString).right.get).map { json =>
+
+    val start = extractField("start", json)
+    val end = extractField("end", json)
+
+    val startDateTime = parseStringToDateTime(start)
+    val endDateTime = parseStringToDateTime(end)
+    if (startDateTime.isAfter(endDateTime) || startDateTime.isEqual(endDateTime))
+      throw new Exception(s"$start is after $end")
```

Comment:
Minor: this exception message would be incorrect if $start == $end. Maybe something more like:

> $start must be before $end

?